### DISCUSSION TITLE
MapLibre Voting Members

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#32] [Governing Board Election]
MapLibre has a Governing Board which consists at the moment of @lseelenbinder, @klokan, @nyurik, and myself (@wipfli). We are in a bootstrapping phase where initially we just volunteered to be members of the Board. But moving forward, we think that it is important that the MapLibre community can elect who serves on its Governing Board. We are therefore preparing an election for the time around FOSS4G which takes places at the end of August 2022.

Inspired by the [meritocratic governance model of Apache](http://oss-watch.ac.uk/resources/meritocraticgovernancemodel) and the [QGIS Charter](https://www.qgis.org/en/site/getinvolved/governance/charter/index.html), we think that the following definition of Voting Members might make sense. Feel free to share your thoughts on this, your feedback is more than welcome!

## Voting Members
Voting Members of the General Meeting are the deciders of policy and any major decisions that affect the general standing. Voting members vote on any motion raised at the Annual General Meeting. Voting members also nominate and elect the Governing Board.

The following people can be Voting Members:

- Whoever contributed in the last two years in a non-trivial way
- Whoever donated (not proportionate to the donation)
- Limit to no more than 3 votes from employees/contractors of a single organization

Voting Members can nominate other community members to receive Voting membership in acknowledgement of their contributions to MapLibre. The Governing Board will vote in private on granting Voting membership to a nominee. The nominee is entitled to request an explanation of any ‘no’ votes against them, regardless of the outcome of the vote.

**Comment by maxammann** [comment]
Sounds good! In my case, I suppose I contributed, but also my current company donated last year.
In this case, I and two other people from my company are allowed to vote, right?

Should we maybe limit the donation to 2-3 years?

**Reply by wipfli** [reply]
Good point about the duration of voting membership after a donation. I think 2 years would be good.

**Reply by wipfli** [reply]
Your company donated to MapLibre which gives one Voting Member. Then you @maxammann contributed in a non-trivial way, which gives a second Voting Member. Now your company could still get a third Voting Member if someone contributed in a non-trivial way, too. Note that the donation alone only qualifies for 1 Voting Member.

**Reply by wipfli** [reply]
At least this would be my guess. Feel free to share counter-opinions...

**Reply by maxammann** [reply]
Makes sense if "Whoever donated" is at maximum one individual.

**Reply by wipfli** [reply]
I think it would be useful if we said that Voting Members are individuals and not companies.

**Comment by ZeLonewolf** [comment]
Do you mean maplibre or the maplibre project?

**Reply by wipfli** [reply]
I guess we should clarify that. @ZeLonewolf what would be your definitions?

**Reply by wipfli** [reply]
...for "MapLibre" and "the MapLibre project"?

**Reply by ZeLonewolf** [reply]
In previous discussions, we've described "maplibre" as referring to the front-end rendering libraries, i.e. maplibre-gl-js and similar related libraries for other platforms, versus the "maplibre project" which is an umbrella organization which may include other software products with different names other than maplibre that are nonetheless part of the rendering stack.  So for example, when talking about contributions, perhaps we might describe "contributions to the maplibre project/org/etc" rather than "contributions to maplibre" which sound like we're specifically referring only to the software carrying the maplibre branding.

**Reply by lseelenbinder** [reply]
I suggest: "Voting Members can nominate other community members to receive Voting membership in acknowledgement of their contributions to the MapLibre-member projects or MapLibre organization."

This opens the door for individuals focused on organization (not just projects / code), but makes it explicit that we're referring to any project under the umbrella (not just rendering libraries with the `maplibre` name).

**Comment by wipfli** [comment]
OSGeo inspiration: https://www.osgeo.org/about/membership-rules/

**Reply by wipfli** [reply]
At OSGeo, the Voting Members can nominate and elect other Voting Members. So there, the Governing Board does not decide who is a Voting Member. I sort of like this concept...

**Comment by maxammann** [comment]
Maybe we should just try this out, but include a duration after which we will reevalute and change the voting guidelines if needed.

How about doing a test run for 2022 and change the rules if needed 2023?

**Reply by nyurik** [reply]
i.e. "if i don't get elected, I don't like it" :)

seriously though, i'm ok with this too

**Reply by maxammann** [reply]
sure, but there needs to be some way to change it :D And it shouldn't be done just by the elected people, but by the community

**Reply by wipfli** [reply]
I agree, @maxammann

**Reply by wipfli** [reply]
I think it would be useful if we had a document where we define how the voting and everything works. If we want to change these definitions, we can, but the Voting Members would have to approve the change with for example a 50 percent majority.

**Reply by wipfli** [reply]
I drafted a MapLibre Charter which is in review at the moment by the Governing Board. My plan is to put the Charter in a markdown file and make a pull request to this repository by the end of the week. Then we can discuss the different points of the Charter in the pull request and finally have a vote on the Charter at the time of FOSS4G...

**Reply by wipfli** [reply]
Ah I forgot that one of the Governing Board members is on vacation at the moment. I will post the MapLibre Charter draft after he is back sometime next week I think.

**Reply by wipfli** [reply]
Here is the Charter draft: https://github.com/maplibre/maplibre/pull/74

**Comment by wipfli** [comment]
And would people be OK if their name or GitHub handle were on a public list of Voting Members or do you rather think that we should keep such a list private?

**Reply by klokan** [reply]
I guess public list of voters is OK

**Reply by wipfli** [reply]
Let's go for a public list of Voting Members. The real name is not necessary I think. The GitHub handle should be enough.

**Reply by wipfli** [reply]
Organization affiliation is not necessary in this public list of Voting Members as long as we know where people come from internally.

**Comment by wipfli** [comment]
What is the minimum donation to qualify as a Voting Member?

**Comment by wipfli** [comment]
Regarding bootstraping: I think the Governing Board should now ask people who they thing would be good Voting Members if they would like to get Voting Membership. This group of people could then do a first round of nominating more Voting Members. And after one round of growing the set of Voting Members, we could have the election of the Governing Board. Would that be appropriate?

**Comment by wipfli** [comment]
If you have any wishes how we should structure the vote and the voting members, please let us know. We plan to have a Governing Board meeting tomorrow and will decide hopefully how to bootstrap this thing...

**Comment by wipfli** [comment]
https://github.com/maplibre/maplibre/issues/36

**Comment by wipfli** [comment]
We had a meeting with David Nalley, president of the Apache Software Foundation and AWS open-source director or so, today. We meaning the Governing Board, i.e. @klokan @nyurik @lseelenbinder and me. I hope I can publish the full notes of the meeting a bit later, but one comment which I really liked was that the exact voting mechanism is not that important. What is important is that things are transparent and understandable for everyone...

**Reply by wipfli** [reply]
Here are some notes of the meeting with David: https://github.com/maplibre/maplibre/discussions/59

**Comment by wipfli** [comment]
In yesterday's steering committee meeting, @mojodna raised the question if non-Voting Members can be on the Governing Board. Feel free to share your thoughts on this...

**Reply by maxammann** [reply]
I think we are fine with keeping the restriction. Basically anyone who wants to join the board could do a symbolic donation of 1$.

The third rule basically avoid that we could have a governing board (5 people) who come from only one company: `Limit to no more than 3 votes from employees/contractors of a single organization`

That means to me that max. 3 out of 5 can theoretically candidate for the board.

**Reply by wipfli** [reply]
Interesting point. I wonder if it would make sense to restrict the number of people from one organization on the Governing Board to 1?

**Reply by flother** [reply]
It would make sense that no single organisation should control the governing board (as in, cannot have more than 50% of the seats). For a five-member board that would mean two seats max.

**Comment by wipfli** [comment]
@JannikGM raised the point that for transparency it might be useful if people could have the option to self-declare who their employer is. What do you think?

**Reply by flother** [reply]
If we're going to limit the number of voting members and governing board members each organisation can have, won't people need to declare their employer?

**Reply by wipfli** [reply]
Yes that sounds correct...

**Comment by sethfitz** [comment]
Disclosure: while I am also personally enthusiastic about MapLibre, AWS currently pays me for time I spend engaging with the project.

Expanding on my question about eligibility of non-Voting Members for the Governing Board from yesterday's meeting, I was thinking about the following scenarios (some/all of which may not be valid; I'm ok with the requirement that candidates come from the voting membership):

1. Someone identifies an awesome outsider who would provide a lot of value to the board and project (through relevant experience in a different area, etc.), but they are not yet a voting member, and might be hard-pressed to become one because they haven't engaged with the project.
2. Someone happens to be employed by a company that's already reached its quota of voting members, but contributes on their own time (or wants to step up their relationship with the project) and would like to run for the board and is unable to.

For reference, eligibility to run for the OpenStreetMap Foundation (and OpenStreetMap US) board requires that you've been a member (paid, in that case, any open to anyway) in good standing for some period of time (also to vote), presumably to mitigate a potential takeover attempt. That seems to work, but the membership is larger and the barrier to becoming a voting member lower.

I think that the OSGeo-style nomination process addresses (1) effectively, as the nomination should have sufficient context for voting members to see why someone would be valuable to the community even if they haven't been directly involved (there are many ways to be involved, which will only increase over time).

One side-effect of working for such a large company (the internet claims Amazon has ~35k software engineers, of which, say, ~25% work for AWS) is that I may see things through a slightly different lens. I posit that there's a distinction between people like me, who are paid to work on MapLibre, and MapLibre contributors who happen to be employed by a company that's otherwise engaged. By limiting the number of voting members per company and requiring that candidates come from that pool, we're potentially acting as gatekeepers for people who may have much to offer us. Gatekeeping and takeover prevention will always be in tension though, so I think we need to accept this somehow.

For example, if @wipfli decided to pick up a side gig delivering packages by balloon for Amazon, he would need to give up his voting membership (or negotiate with an existing member to trade slots). To me, this is decidedly different than AWS hiring someone from the community to work specifically on MapLibre.

To be clear, I think we are right to consider the threat of undue corporate influence or takeover (and if you're ever concerned that you're seeing this from Amazon, please reach out to me directly), but I think there's some nuance here that we haven't yet explored.

In the interest of offering solutions, I have 2 proposals for voting membership and +1 to @flother's "no single organization may control more than 50% of the board seats":

1. The per-company cap should be proportional to the size of the voting membership (~10%: 3 against 34 total today) so that community growth can be inclusive of those affiliated with companies who are involved.
2. Distinguish between "community" involvement (on one's own time, unpaid, with no discernible link to one's day job; we can come up with examples) and "professional" involvement (paid, at least partially, and linked to one's day job). "Professional" involvement would count toward the per-company cap, "community" would not.

**Reply by wipfli** [reply]
That's a great idea for a side gig...

**Reply by wipfli** [reply]
And I like the concept of paid vs unpaid contributions and associated limits.

**Reply by lseelenbinder** [reply]
@mojodna Just wanted to say I really appreciate your perspective, and I think your proposals are quite logical!

> The per-company cap should be proportional to the size of the voting membership (~10%: 3 against 34 total today) so that community growth can be inclusive of those affiliated companies who are involved.

100% agreed. I actually proposed this at some point as one solution to the obvious problem of a hard cap, so it's nice to see another community member come up with the same numbers

> Distinguish between "community" involvement (on one's own time, unpaid, with no discernible link to one's day job; we can come up with examples) and "professional" involvement (paid, at least partially, and linked to one's day job). "Professional" involvement would count toward the per-company cap, "community" would not.

I agree in principle. The one issue in my mind here is the impossibility to disentangle someone as a member MapLibre "on the side" while being full-time at a large organization. If said organization wants to exert undue power over MapLibre, all they need to do is compare voting member roles to employees and twist some arms. To be clear, I'm not saying AWS / Amazon or any other organization we work with would do this, but it's seems entirely plausible that it could happen.

One way to counteract this is by having a set cap on the governing board membership (e.g., one person per organization of any size), whether or not that individual's contributions are on the side or not. This means one organization could never fully control decisions, regardless of the composition of the voting membership.

Thoughts?

**Reply by sethfitz** [reply]
> One way to counteract this is by having a set cap on the governing board membership (e.g., one person per organization of any size), whether or not that individual's contributions are on the side or not. This means one organization could never fully control decisions, regardless of the composition of the voting membership.

Yes, agreed. This was implicit in my agreement with @flother's "no single organization may control more than 50% of the board seats", but best to be explicit. I don't know if 50% is the right number, but I agree with the intent, and that it should be inclusive of "community" involvement (large companies already have hurdles for new employees with board seats to deal with w/r/t conflicts of interest, so I don't see such a MapLibre requirement as an additional burden).

**Comment by wipfli** [comment]
Election Timeline:

* Voting Members can nominate other people to become Voting Members until **2022-08-11**, see https://github.com/maplibre/maplibre/issues/71
* People can submit a candidacy for serving on the Governing Board until **2022-08-11**, see https://github.com/maplibre/maplibre/issues/70
* Voting Members vote on granting Voting Membership to the new people on **2022-08-18**
* The Governing Board election will take place **2022-08-25**
